c7n/resources/ebs.py

```diff
@@ -50,27 +51,59 @@ class SnapshotAge(AgeFilter):
         days={'type': 'number'},
         op={'type': 'string', 'enum': OPERATORS.keys()})
     date_attribute = 'StartTime'
+    
+
+def _filter_ami_snapshots(self, snapshots):
+# gets a listing of all AMI snapshots and compares resources to the list
+    c = local_session(self.manager.session_factory).client('ec2')
+    ami_snaps = []
+    for i in c.describe_images(Owners=['self'])['Images']:
```

Comment:
@ocampocj: Take a look at `UnusedLaunchConfig.process()` in `asg.py` for an example of querying the cache. You could do something like:
```
        amis = self.manager._cache.get(
            {'region': self.manager.config.region,
             'resource': 'ami'})
        if amis is None:
            # This will populate the cache.
            ami_manager = AMI(self.manager.ctx, {})
            amis = ami_manager.resources()
```